src/main/java/de/tum/in/www1/artemis/service/connectors/bitbucket/BitbucketService.java

```diff
@@ -807,6 +812,21 @@ public Commit getLastCommitDetails(Object requestBody) throws BitbucketException
         return commit;
     }
 
+    @Override
+    public ZonedDateTime getPushDate(ProgrammingExerciseParticipation participation, String commitHash) {
+        final var url = bitbucketServerUrl + "/rest/api/latest/projects/" + participation.getProgrammingExercise().getProjectKey() + "/repos/"
+                + urlService.getRepositorySlugFromRepositoryUrl(participation.getVcsRepositoryUrl()) + "/ref-change-activities?start=0&limit=25";
+        final var response = restTemplate.exchange(url, HttpMethod.GET, null, BitbucketChangeActivitiesDTO.class);
+        if (response.getStatusCode() != HttpStatus.OK || response.getBody() == null) {
+            throw new BambooException("Unable to get push date for participation " + participation.getId() + "\n" + response.getBody());
+        }
+        final var changeActivities = response.getBody().getValues();
+
+        final var activityOfPush = changeActivities.stream().filter(activity -> activity.getRefChange().getToHash().equals(commitHash)).findFirst()
```

Comment:
```suggestion
        final var activityOfPush = changeActivities.stream().filter(activity -> commitHash.equals(activity.getRefChange().getToHash())).findFirst()
```

Without knowing the response from Bamboo, this may be a more robust way.